### DISCUSSION TITLE
What is the right way to patch "${TMPDIR}/work-shared/L4T-${L4T_BSP_ARCH}-${PV}-${PR}/Linux_for_Tegra" sources

### DISCUSSION THREAD

**Discussion by aurelien-enchanted-tools** [OE4T/meta-tegra#1430] [Q&A] [answered]
Hello World,

branch mickledore

I tried to patch `./bootloader/t186ref/BCT/tegra234-mb2-bct-scr-p3767-0000.dts` and `./bootloader/t186ref/BCT/tegra234-mb2-bct-misc-p3767-0000.dts` with a `tegra-bootfiles_35.4.1.bbappend` :
```
FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"

SRC_URI:tegra234 += "\
    file://0001-Disable-EEPROM.patch \
    file://0002-Adaptation-to-the-Carrier-Board-with-HDMI-for-the-Or.patch \
    "

addtask do_patch before do_compile
```
But it is not stable. What is the right way to do this? Thank you in advance.

**Comment by madisongh** [comment, marked as answer]
If you really need to apply patches to anything in the BSP, apply them in a  bbappend to the `tegra-binaries` recipe.

**Comment by dwalkes** [comment]
See potentially related discussion in https://github.com/OE4T/meta-tegra/discussions/1094

**Comment by tjoslin** [comment]
@aurelien-enchanted-tools @dwalkes did you end up going forward with one of these approaches? I'm attempting to apply the same HDMI carrier board patch from the l4t documentation

**Reply by dwalkes** [reply]
@tjoslin I used the mechanism described in https://github.com/OE4T/meta-tegra/discussions/1094#discussioncomment-3863437
